apollo-router-core/src/layers/instrument.rs

```diff
@@ -0,0 +1,73 @@
+use futures::future::BoxFuture;
```

Comment:
After the plugin utils etc stuff merges, you'll see the minimal style of documentation that is required. You can lift some of the documentation from that PR and use it to document the new components (such as this instrument layer/service) that you are adding.

Reply:
I've added some minimal docs, I'm happy to revisit in a larger docs effort.